component/custom/custom_loader.go

```diff
@@ -0,0 +1,40 @@
+package custom
+
+import (
+	"os"
+	"plugin"
+	"strings"
+)
+
+func init() {
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		return
+	}
+
+	// environment variable can contain multiple paths separated by colon.
+	paths := strings.Split(pluginPath, ":")
+
+	for _, file := range paths {
+		println("custom_loader::Opening plugin:", file)
+		p, err := plugin.Open(file)
+		if err != nil {
+			println("custom_loader::Error opening plugin:", err.Error())
+			return
+		}
+		println("custom_loader::Plugin opened successfully")
+
+		initFunc, err := p.Lookup("InitPlugin")
```

Comment:
>initFunc, err := p.Lookup("InitPlugin") [](http://example.com/codeflow?start=2&length=39)

This InitPlugin the only thing that we can validate ? Is this the only required method, if no, should we validate presence of those aswell ?

Reply:
What hari suggested to use i.e GetCustomComponent instead of initPluging method should handle the validation whether the custom component implements the BaseComponent or not.
Added in new iteration.